File: Qytetet Java/src/modeloqytetet/PruebaQytetet.java

//package modeloqytetet;
//import java.util.ArrayList;
//import java.util.Scanner;
//
///**
// *
// * @author miguel
// */
//public class PruebaQytetet {
//    private static Qytetet juego = Qytetet.Instance();
//    static Tablero tablero = new Tablero();
//    private static final Scanner in =new Scanner (System.in);
//    /**
//     * @param args the command line arguments
//     */
//    private static ArrayList<Sorpresa> SorpresasPositivas (ArrayList<Sorpresa> mazo){
//        ArrayList<Sorpresa> positivo = new ArrayList<>();
//        for (Sorpresa s : mazo){
//            if(s.getValor()>0)
//                positivo.add(s);
//        }
//        return positivo;
//    }
//    
//    private static ArrayList<Sorpresa> SorpresasCasillas (ArrayList<Sorpresa> mazo){
//        ArrayList<Sorpresa> casillas = new ArrayList<>();
//        for (Sorpresa s : mazo){
//            if(s.getTipo()==TipoSorpresa.IRACASILLA)
//                casillas.add(s);
//        }
//        return casillas;
//    }
//    
//    private static ArrayList<Sorpresa> SorpresasTipo (ArrayList<Sorpresa> mazo, TipoSorpresa tipo){
//        ArrayList<Sorpresa> tipoSorpresa = new ArrayList<>();
//        for (Sorpresa s : mazo){
//            if(s.getTipo()==tipo)
//                tipoSorpresa.add(s);
//        }
//        return tipoSorpresa;
//    }
//    
//    private static ArrayList<String>  getNombreJugadores(){
//        int numeroJugadores;
//        ArrayList<String> nombres = new ArrayList<>(); 
//        System.out.println("Introduzca el numero de jugadores (entre 2 y 4)");{
//            do{
//                numeroJugadores = in.nextInt();
//            }while (numeroJugadores < 2 && numeroJugadores > 4);
//        }
//        for (int i=0; i < numeroJugadores ;i++){
//            System.out.println("Introduzca el nombre del jugador "+i);
//            String s = in.next();
//            nombres.add(s);
//        }
//          return nombres;  
//    }
//    
//    public static void main(String[] args) {
//        ArrayList <String> nombres = new ArrayList<String>();
//        nombres=getNombreJugadores();
//        juego.inicializarJuego(nombres);
//        juego.getMazo().toString();
//               
//        for(int i=0;i<9;i++){
//            
//            System.out.println(juego.getMazo().get(i));
//        }
//        tablero.getCasillas().toString();
//        for(int i=0;i<20;i++){
//            System.out.println(tablero.getCasillas().get(i));
//            
//            }               
//        System.out.println("Los jugadores  son: ");
//        System.out.println(juego.getJugadores().toString()); 
//
//        System.out.println("La unica instacia de Qytetet:");
//        System.out.println(juego);
//       //Vamos a probar lo implementado en la práctica 3
//        juego.mover(3);
//        juego.getJugadorActual().comprarTituloPropiedad();
//        juego.siguienteJugador();
//        juego.mover(3);
//        System.out.println(juego.getJugadorActual().getCasillaActual());
//        System.out.println(juego.getJugadorActual().getSaldo());
//        // Vamos a caer en todas las Sorpresas
//       juego.jugar();
//       juego.mover(2);
//       System.out.println(juego.getJugadorActual().getCasillaActual());
//       juego.aplicarSorpresa();
//       System.out.println(juego.getJugadorActual().getCasillaActual());
//       juego.mover(7);
//       System.out.println(juego.getJugadorActual().getCasillaActual());
//       juego.aplicarSorpresa();   
//       juego.mover(12);
//       System.out.println(juego.getJugadorActual().getCasillaActual());
//       juego.aplicarSorpresa();
//       juego.siguienteJugador();
//       juego.hipotecarPropiedad(3);
//       juego.cancelarHipoteca(3);
//       juego.venderPropiedad(3);
//       juego.mover(4);
//       juego.comprarTituloPropiedad();
//       juego.edificarCasa(4);
//       juego.edificarCasa(4);
//       juego.edificarCasa(4);
//       juego.edificarCasa(4);
//       juego.edificarHotel(4);
//       System.out.println(juego.getJugadorActual().getCasillaActual().getTitulo().getNumCasas());
//       System.out.println(juego.getJugadorActual().getCasillaActual().getTitulo().getNumHoteles());
//       juego.hipotecarPropiedad(juego.getJugadorActual().getCasillaActual().getNumeroCasilla());
//       //juego.siguienteJugador();
//       //juego.getJugadorActual().irACarcel(juego.obtenerCasillasTablero().get(5));
//       System.out.println(juego.getJugadorActual().getCasillaActual());
//       /*if(juego.getJugadorActual().tengoCartaLibertad()){
//           juego.intentarSalirCarcel(MetodoSalirCarcel.PAGANDOLIBERTAD);
//       }       
//       else{
//           boolean salirDeLaCarcel = juego.intentarSalirCarcel(MetodoSalirCarcel.TIRANDODADO);           
//           if(salirDeLaCarcel)
//               System.out.println("Consigue salir de la cárcel");     
//           else
//               System.out.println("Al palo");       
//        }
//       juego.jugar();
//       System.out.println(juego.getJugadores());
//       juego.getJugadorActual().modificarSaldo(-300);
//       juego.siguienteJugador();
//       juego.getJugadorActual().modificarSaldo(300);
//       juego.obtenerRanking();
//       System.out.println(juego.getJugadores());
//       System.out.println("Vamos a probar la practica 4");
//       juego.mover(2);
//       System.out.println(juego.getJugadorActual().getCasillaActual());
//       juego.aplicarSorpresa();
//       juego.siguienteJugador();
//       juego.mover(7);
//       System.out.println(juego.getJugadorActual().getCasillaActual());
//       juego.aplicarSorpresa();
//       System.out.println(juego.getJugadores());*/
//        }
//    }
//

File: Qytetet Ruby/lib/prueba_qytetet.rb
##encoding: utf-8
#require_relative "sorpresa"
#require_relative "qytetet"
#require_relative "tipo_sorpresa"
#require_relative "tablero"
#require_relative "casilla"
#require_relative "metodo_salir_carcel"
#require_relative "especulador"
#
#module ModeloQytetet
#  
#  class PruebaQytetet
#    @@juego=Qytetet.instance 
# 
#    def self.mayor_cero()
#      mayor_cero=Array.new
#      for s in @@juego.mazo.each
#        if s.valor>0
#          mayor_cero<<s      
#        end
#      end
#    return mayor_cero
#    end
#    
#    def self.ir_a_casilla()
#      ir_a_casilla=Array.new
#      for s in @@juego.mazo.each
#        if(s.tipo==:Ir_a_casilla)
#          ir_a_casilla<<s
#        end
#        
#      end
#     return ir_a_casilla
#    end
#    def self.tipo_sorpresa (sorpresa)
#      tipo_sorpresa=Array.new
#      for s in @@juego.mazo.each
#        if s.tipo==sorpresa
#          tipo_sorpresa<<s
#        end
#        
#      end
#      return tipo_sorpresa
#    end
#    def self.get_nombres_jugadores()
#      puts "Introduzca el numero de jugadores"
#      numero=gets.chomp.to_i
#      nombres=Array.new
#      
#      for i in 1..numero
#        puts "Introduzca el nombre del jugador #{i}"
#        nombres.push(gets.chomp)
#      end      
#      return nombres
#    end
#    
#    def self.main
#      tablero=Tablero.new
#      nombres = get_nombres_jugadores
#      @@juego.inicializar_juego(nombres)
#      @@juego.mover(7)
##      @@juego.comprar_titulo_propiedad
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.mover(7)
#      puts @@juego.jugador_actual
#      puts @@juego.carta_actual
#      @@juego.aplicar_sorpresa
##      puts @@juego.carta_actual
##      puts @@juego.jugadores
##      @@juego.mover(3)
##      puts @@juego.jugador_actual.casilla_actual
##      @@juego.comprar_titulo_propiedad
#      puts @@juego.jugador_actual
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_hotel(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_hotel(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_hotel(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_hotel(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      puts @@juego.jugador_actual
##      @@juego.siguiente_jugador
##      @@juego.mover(3)
##      puts @@juego.jugadores
##      @@juego.mover(3)
##      @@juego.jugador_actual.comprar_titulo_propiedad()
##      @@juego.siguiente_jugador
##      @@juego.mover(3)
##      puts @@juego.jugadores
##      @@juego.siguiente_jugador
##      # Probamos a caer en una sorpresa y a aplicar sorpresa después
##      @@juego.mover(2)
##      @@juego.aplicar_sorpresa
##      puts @@juego.jugador_actual.casilla_actual
##      @@juego.mover(3)
##      @@juego.hipotecar_propiedad(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.cancelar_hipoteca(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.vender_propiedad(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_casa(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##      @@juego.edificar_hotel(@@juego.jugador_actual.casilla_actual.numeroCasilla)
##       #Probamos a salir de la carcel
##      @@juego.jugador_actual.ir_a_carcel(tablero.carcel)
##      if(@@juego.jugador_actual.tengo_carta_libertad)
##        @@juego.intentar_salir_carcel(MetodoSalirCarcel::PAGANDOLIBERTAD)
##      else
##        consigue = @@juego.intentar_salir_carcel(MetodoSalirCarcel::TIRANDODADO)
##
##        if consigue
##          puts "Sales de la cárcel"
##        else
##          puts "NO sales de la carcel"
##        end
##      end
#    end
#    
#end
#PruebaQytetet.main
#end